architecture/core.rst

```diff
@@ -412,10 +412,11 @@ Interrupt management resources are declared in header file
 kernel/interrupt.h_. Interrupt consumers main API functions are:
 
     - ``interrupt_enable()`` and ``interrupt_disable()`` to respectively
-      enable and disable an interrupt.
+      enable or disable an interrupt.
 
     - ``interrupt_mask()`` and ``interrupt_unmask()`` to respectively
-      mask/unmask an interrupt.
+      mask or unmask an interrupt. Masking an interrupt keeps it enabled
+      but disables the asynchronus notification of the event.
```

Comment:
That's not quite true. They are actually identical, mask/unmask are implemented using disable/enable in all upstream drivers. How about something like:

Masking of an enabled interrupt temporarily disables the interrupt while unmasking enables a previously masked interrupt. `interrpt_mask()` and `interrpt_unmask()` are allowed to be called from an interrupt context, but `interrupt_enable()` and `interrupt_disable()` not so.